drake/systems/plants/RigidBodyTree.h

```diff
@@ -286,35 +286,44 @@ class DRAKERBM_EXPORT RigidBodyTree
   Eigen::Matrix<typename DerivedV::Scalar, Eigen::Dynamic, 1> frictionTorques(Eigen::MatrixBase<DerivedV> const & v) const;
 
   template <typename Scalar, typename DerivedPoints> // not necessarily any relation between the two; a major use case is having an AutoDiff KinematicsCache, but double points matrix
-  Eigen::Matrix<Scalar, Eigen::Dynamic, DerivedPoints::ColsAtCompileTime> forwardKin(
-      const KinematicsCache<Scalar> &cache, const Eigen::MatrixBase<DerivedPoints> &points, int current_body_or_frame_ind, int new_body_or_frame_ind, int rotation_type) const
+  Eigen::Matrix<Scalar, 3, DerivedPoints::ColsAtCompileTime> transformPoints(
+      const KinematicsCache<Scalar> &cache, const Eigen::MatrixBase<DerivedPoints> &points, int from_body_or_frame_ind, int to_body_or_frame_ind) const
   {
-    cache.checkCachedKinematicsSettings(false, false, "forwardKin"); // rely on forwardJac for gradient cache check
+    static_assert(DerivedPoints::RowsAtCompileTime == 3 || DerivedPoints::RowsAtCompileTime == Eigen::Dynamic, "points argument has wrong number of rows");
+    auto T = relativeTransform(cache, to_body_or_frame_ind, from_body_or_frame_ind);
```

Comment:
Yeah, good idea to combine those method signature changes. But I would prefer to not make all of those changes in this pull request.